internal/service/report.go

```diff
@@ -0,0 +1,572 @@
+package service
+
+import (
+	"bytes"
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/kubev2v/migration-planner/api/v1alpha1"
+	"github.com/kubev2v/migration-planner/internal/store/model"
+)
+
+type ReportService struct {
+}
+
+func NewReportService() *ReportService {
+	return &ReportService{}
+}
+
+func (r *ReportService) GenerateReport(source *model.Source, options ReportOptions) (string, error) {
```

Comment:
I suggest also to:

- Create to subdirectories here (csv and html)
- seperate the code to the relevant directories
- Instead of using here switch/case, use one common intrface which implemented by both

I think this will make the code more clear and easier to add other types in the future
What do you think?